src/RageSoundPosMap.cpp

```diff
@@ -145,7 +137,7 @@ std::int64_t pos_map_queue::Search( std::int64_t iSourceFrame, bool *bApproximat
 			/* iSourceFrame lies in this block; it's an exact match.  Figure
 			 * out the exact position. */
 			int iDiff = int(iSourceFrame - pm.m_iSourceFrame);
-			iDiff = std::lrint( iDiff * pm.m_fSourceToDestRatio );
+			iDiff = static_cast<int>(std::floor(( iDiff * pm.m_fSourceToDestRatio )));  // avoid lrint for best performance on windows
```

Comment:
> Why do we need a std::floor if we're already casting? It seems redundant so I would remove.

As mentioned above, adding the `std::floor` will ensure the behavior is exactly the same as `lrint()`. I have not tested extensively to see if there is actually any difference in practice, but it should be fine to just use `static_cast<int>()`.

> Also if the iSourceFrame is a std::int64_t, aren't we losing precision when having a narrowing conversion to int? Is that fine here because we expect it to be small?

I haven't noticed any difference in stability whether this is an `int` or `int64_t`, but now that I think about it, for the sake of keeping everything 64-bit wide I will make this an `int64_t`

Reply:
????

`static_cast<int>(std::floor(d))` definitely does not have the same behavior as `std::lrint`?

`std::floor` essentially truncates similarly to to `static_cast<int>` which is super easy to test (which I did here): https://ideone.com/dkBJ4w

Also I was more asking about the difference in theory instead of practice. You're definitely more familiar with this area of the code than I am so I figured you would knew the potential consequences. If we wanted more similar behavior as `lrint` for positive numbers (while not exact) we could do something like `static_cast<int>(foo + 0.5)`, but wondering if that' s even needed.

Reply:
> static_cast<int>(std::floor(d)) definitely does not have the same behavior as std::lrint?

Absolutely correct, that's my oversight. I should have written `static_cast<int>(std::round(d))` instead. The +0.5 is a great idea too.

> Also I was more asking about the difference in theory instead of practice.

I see what you were asking now. I don't believe that it's critical that the value here is exactly what `std::lrint` provides. While it's possible `std::lrint` was chosen specifically for this routine, if the rounding is off by a single digit, that's just a single frame of audio of incorrectness. We'll still be landing close enough to the proper position that everything should keep working fine here. This code is not going to execute fast enough that we beat a single audio frame from being processed; landing in a good position in the buffer is "good enough".